scan_to_inventory/static/www/models/m_product_product.js

```diff
@@ -0,0 +1,18 @@
+angular.module('scan_to_inventory').factory(
+        'ProductProductModel', [
+        '$q', '$rootScope', 'jsonRpc',
+        function ($q, $rootScope, jsonRpc) {
+
+    return {
+        LoadProductList: function() {
+            return jsonRpc.call(
+                    'product.product', 'scan_to_inventory_load_product', []).then(function (res) {
+                $rootScope.ProductListByEan13 = res;
+                var quantity = 0;
+                for (var a in res){quantity ++;}
```

Comment:
From your log trace : res is an object with a number as key and an object as value.
Size of an object does not make a lot of sense in your case. It's may be better to transfert `res` as an array.

What you want to do is : 
`Object.keys(res).length`